src/layers/perfetto.rs

```diff
@@ -97,24 +97,42 @@ impl Layer {
     /// - `PERFETTO_BUFFER_SIZE_KB`: size of the buffer in kilobytes. Default: 50 * 1024. Is used only with the in-process backend.
     /// - `PERFETTO_PLATFORM_NAME`: custom platform name. Default: architecture of the CPU that is currently in use.
     pub fn new_from_env() -> Result<(Self, PerfettoGuard), perfetto_sys::Error> {
-        let trace_file_patch = match std::env::var("PERFETTO_TRACE_FILE_PATH") {
-            Ok(path) => path,
-            Err(_) => {
-                let time = get_formatted_time();
-                let branch = get_current_branch_revision();
-                let platform = std::env::var("PERFETTO_PLATFORM_NAME")
-                    .unwrap_or(std::env::consts::ARCH.to_string());
-                let hostname = gethostname().into_string();
-
-                format!(
-                    "{}{}{}{}.perfetto-trace",
-                    time,
-                    branch.map_or(String::new(), |b| format!("-{}", b)),
-                    format!("-{}", platform),
-                    hostname.map_or(String::new(), |h| format!("-{}", h)),
-                )
+        let (timestamp_filename, timestamp_iso) = get_formatted_time();
+        let git_info = get_git_info();
+
+        // Compute the trace-file path:
+        //    - If PERFETTO_TRACE_FILE_PATH is set, use it verbatim.
+        //    - Otherwise generate
+        //        <timestamp>-<branch>-<commit_hash>[-dirty]-<platform>-<hostname>.perfetto-trace
+        //      and, if PERFETTO_TRACE_DIR is set, prepend that directory.
+        let output_path = if let Ok(p) = std::env::var("PERFETTO_TRACE_FILE_PATH") {
+            std::path::PathBuf::from(p)
+        } else {
+            let mut parts = Vec::with_capacity(6);
+            parts.push(timestamp_filename);
+            if let Some(g) = &git_info {
+                parts.push(sanitize_filename(&g.branch));
+                parts.push(g.commit_short.clone());
+                if !g.is_clean {
+                    parts.push("dirty".to_string());
+                }
             }
+            let platform = std::env::var("PERFETTO_PLATFORM_NAME")
+                .unwrap_or_else(|_| std::env::consts::ARCH.to_string());
+            parts.push(platform);
+            let hostname = gethostname().to_string_lossy().to_string();
+            parts.push(hostname);
+
+            let fname = format!("{}.perfetto-trace", parts.join("-"));
+
+            let dir = std::env::var("PERFETTO_TRACE_DIR").unwrap_or_else(|_| ".".to_string());
+            std::path::PathBuf::from(dir).join(fname)
         };
+        let output_path_str = output_path.to_string_lossy().to_string();
+
+        // Record the chosen path for external scripts
+        std::fs::write(".perfetto_trace_path", &output_path_str)
+            .expect("Unable to write .perfetto_trace_path");
```

Comment:
1. Better propagate error instead of `expect`, just `?` should be enough
2. [minor] Maybe name the file `.last_perfetto_trace_path`?